MHub / GC - Add GC nnUNet Pancreas model

silvandeleemput opened the pull request:
This PR contains the required code for the GC nnUNet Pancreas model.
GitHub Repo: https://github.com/DIAGNijmegen/CE-CT_PDAC_AutomaticDetection_nnUnet
GC page: https://grand-challenge.org/algorithms/pdac-detection/

### Algorithm I/O
  * Input should be a CT Abdomen image (MHA) (This MHub implementation expects a Dicom which is converted to MHA using MhaPanImgConverter)
  * Outputs are:
    * MHA and DICOMSEG Segmentation with: 1-veins, 2-arteries, 3-pancreas, 4-pancreatic duct, 5-bile duct, ~6-cysts, 7-renal vein~
    * A heatmap of the pancreatic tumor likelihood
    * Case-level pancreatic tumor likelihood value, in range [0, 1]
  * Test images for this algorithm can be downloaded from here: https://zenodo.org/record/3431873
  * IDC test image:
```
Dataset: CTpred-Sunitinib-panNET

Case: PAN_14

SeriesInstanceUID: 1.3.6.1.4.1.14519.5.2.1.98806944896875131873924657854330253112

s5cmd --no-sign-request --endpoint-url https://s3.amazonaws.com cp 's3://idc-open-data/5475d74e-33db-497e-ace0-00faba1a6a9a/*' .
```  
### Caveats
* ~The PR target is `main`, but should be something like `m-gc-nnunet-pancreas`~
* ~In the Dockerfile the MHub model repo integration is currently marked `TODO` since it requires the creation of the appropriate branch for this code first.~
* ~This implementation still uses a `run.py` script because the MHAConverter doesn't have the panimg backend yet.~
* ~I added a temporary `HEATMAP = Meta(mod="heatmap")` which might be moved to `MHubio/core/templates.py`~
* ~The segmentation regions for this algorithm are not yet part of dseg repo, hence I have added a dseg.json file~

silvandeleemput committed:
WIP initial pancreas_pdac model commit

silvandeleemput committed:
renamed nnunet_pancreas_pdac -> gc_nnunet_pancreas, addded run script, mhaimporter and specific gcnnunetpancreasrunner

* Fixed Dockerfile with nocuda for now using the new base image
* Added two utils for this runner (MHAImporter, custom GCnnunetPancreasRunner)
* Marked bunch of things as TODO

silvandeleemput committed:
update for new base image, add dseg.json for class labels

silvandeleemput committed:
Merge branch 'main' into m-gc-nnunet-pancreas

silvandeleemput committed:
cleanup code runner and run, configure dsegconverter, dataorganizer

LennyN95 commented:
Hi Sil,

Great work!

Some comments:
- We changed our strategy and ditch the model branches. We'll still use those for creation/patching of models but in the Docker container you can fetch the model from the `main` branch.
- You can use the run module instead of the scripts for custom modules (like converters) as long as they're placed in the utils folder and the script name matches the class name

We'll inspect the segmentations and update them to SegDB

silvandeleemput commented:
> * We changed our strategy and ditch the model branches. We'll still use those for creation/patching of models but in the Docker container you can fetch the model from the `main` branch.

Great, I'll target the `main` branch instead.

> * You can use the run module instead of the scripts for custom modules (like converters) as long as they're placed in the utils folder and the script name matches the class name

Ah, I found that while the Runner modules get automatically picked up, this doesn't happen with converters (I added it as a `*` import to the `__init__.py`. Maybe you can check this? It might have something to do with `Module` class versus `Converter` class.

LennyN95 commented:
Converters will be picked up automatically only if the file and class names are identical (which implies a one-module per class policy).

silvandeleemput commented:
> Converters will be picked up automatically only if the file and class names are identical (which implies a one-module per class policy).

Okay, that makes sense. On a related note, what is the current status of using `panimg` as a backend for conversion using the MhaConverter?

LennyN95 commented:
I just updated our MHA conversion module. You now can specify `engine: plastimatch|panimg` in the config file to use panimg as conversion backend ;)

```
  MhaConverter:
    engine: plastimatch
```

silvandeleemput committed:
add panimg backend for mhaconverter and cleanup

silvandeleemput committed:
Merge branch 'MHubAI:main' into m-gc-nnunet-pancreas

silvandeleemput committed:
Updated and cleaned Dockerfile and Runner and added some comments

silvandeleemput commented:
I have updated this PR:
* Uses the panimg backend now for the MhaConverter
* Repository version has been pinned to a commit hash
* Added more comments for clarity

Some open issues:
* [x] In the Dockerfile the MHub model repo integration is currently marked `TODO` since this still has to be tested after integration
* ~A `HEATMAP = Meta(mod="heatmap")` is still marked as TODO in the runner file which should probably be moved to `MHubio/core/templates.py`~
* [x] The segmentation regions for this algorithm are not yet part of dseg repo, hence I have added a dseg.json file, will these be added?

silvandeleemput committed:
Merge branch 'MHubAI:main' into m-gc-nnunet-pancreas

silvandeleemput committed:
add meta.json

LennyN95 commented:
The algorithm delineates the following ROI.

_**NOTE:** We will add them to our SegDB shortly. Some need clarification (what is delineated, in what context, complete/incomplete, etc.). A visual demonstration would be ideal to ensure correct usage of SCT codes._ 

- [ ] Veins _(needs clarification)_
- [ ] Artery, right and left _(needs clarification)_
- [x] Pancreas _(SegDB ID `PANCREAS`)_
- [ ] Pancreatic duct _(SCT 69930009)_
- [ ] Bile duct _(SCT 28273000)_
- [x] Cysts _(SegDB ID `CYST`; usage in context, e.g. pancreatic cyst: `PANCREAS+CYST`)_
- [ ] Renal vein _(SCT 56400007)_

LennyN95 commented on models/gc_nnunet_pancreas/dockerfiles/Dockerfile:
```diff
@@ -0,0 +1,46 @@
+    mv /opt/algorithm/nnUNetTrainerV2_Loss_CE_checkpoints.py "$SITE_PKG/nnunet/training/network_training/nnUNetTrainerV2_Loss_CE_checkpoints.py"
+
+# Clone the main branch of MHubAI/models TODO check if ok
+RUN git stash \
```
This code is only functional upon approval and integration in the mhubai/models repository. 
To avoid this and to allow for testing without modifications, please use the recently introduced model import build utility instead:

```Dockerfile
# Import the MHub model definiton
ARG MHUB_MODELS_REPO
RUN buildutils/import_mhub_model.sh gc_nnunet_pancreas ${MHUB_MODELS_REPO}
```

silvandeleemput replied:
Done

LennyN95 commented:
The algorithm delineates the following ROI.

_**NOTE:** We will add them to our SegDB shortly. Some need clarification (what is delineated, in what context, complete/incomplete, etc.). A visual demonstration would be ideal to ensure correct usage of SCT codes._ 

- [x] Veins _(needs clarification)_
- [x] Artery, right and left _(needs clarification)_
- [x] Pancreas _(SegDB ID `PANCREAS`)_
- [x] Pancreatic duct _(SCT 69930009)_
- [x] Bile duct _(SCT 28273000)_
- [x] Cysts _(SegDB ID `CYST`; usage in context, e.g. pancreatic cyst: `PANCREAS+CYST`)_
- [x] Renal vein _(SCT 56400007)_

silvandeleemput committed:
Merge branch 'MHubAI:main' into m-gc-nnunet-pancreas

silvandeleemput committed:
update mhub model definition import Dockerfile

silvandeleemput committed:
removed first comment line in Dockerfile

silvandeleemput commented:
Some of the algorithm segmentation output viewed on GC:
![nnunet_pancreas_seg_labels](https://github.com/MHubAI/models/assets/3715102/a0a5544c-31e1-420f-9b2f-964fbd90633c)

I think that for both the veins and for the arteries there is no distinction between left and right, just one broad class for each.
Veins and arteries seem incomplete in the sagittal direction (top and bottom).

The area of focus is the pancreas enclosed in a bounding box with some margins for the other tissues.

LennyN95 commented:
Not sure if there is (great) value in adding the incomplete structures, even on a general SCT. 
I'd focus on the complete ones and regard the others (they will still be accessible through the original model output files). 

@fedorov what do you think?

LennyN95 commented:
Added the following IDs to SegDB: `PANCREATIC_DUCT`,  `BILE_DUCT`,  `RENAL_VEIN`.

fedorov commented:
What is the definition of "incomplete structure" anyway? Complete or incomplete requires someone to define the reference. 

I think whatever outputs model produces, MHub should keep. And then someone else sometime later can come up with a model that classifies "complete vs incomplete" structures.

silvandeleemput committed:
Added segdb export, removed dseg.json, added remapped output to runner, updated Dockerfile and config

silvandeleemput commented:
### Update

For now, I have removed the dseg.json and added the DSeg labels as suggested by @LennyN95. without the veins and arteries. It might be nice to add them later though, as it would simplify the runner code by removing the need for a remapping. 

Regarding the incomplete segmentations: The algorithm segments all structures in a small area around the pancreas within a bounding box established from the input image. However, it is later positioned back into an image with the resolution of the original image yielding large zero-padded areas at the borders. For at least the veins and arteries this results in "incomplete" structures when transitioning from the center to the zero-padded areas.

I successfully used the following IDC CT image for testing the model:
```
Dataset: CTpred-Sunitinib-panNET

Case: PAN_01

Series Instance UID: 1.3.6.1.4.1.14519.5.2.1.72227081914373684637438450754406716260

Giving me the following outputs:

└── 1.3.6.1.4.1.14519.5.2.1.72227081914373684637438450754406716260
    ├── [854K]  nnunet_pancreas_heatmap.mha
    ├── [6.2M]  nnunet_pancreas.seg.dcm
    └── [123K]  nnunet_pancreas.seg.mha
```

And the output looks, something like the following:
![image](https://github.com/MHubAI/models/assets/3715102/c728d64f-7e2e-4f24-8436-8a1fbea888ef)
From left to right: Input CT image, Heatmap (nnunet_pancreas_heatmap.mha), Pancreas segmentation map (nnunet_pancreas.seg.mha)

LennyN95 commented:
I successfully run the implementation, yay :)

The pancreas looks a bit "wild" though, @silvandeleemput @fedorov ?
![Bildschirmfoto 2023-12-07 um 14 42 58](https://github.com/MHubAI/models/assets/14541510/4ecab1e1-334b-4a5f-8a14-588410cbab03)

Some issues:
- typing_extensions conflict needs to be addressed
- polluted console out (similar to PICAI); this should be addressed for all algorithms (before we enter the testing phase).
- for final testing, we need your output directory (zipped) and at least one case for each of the segmented structures (the current one only shows the pancreas, pancreatic duct, and bile duct).

silvandeleemput commented:
> I successfully run the implementation, yay :)

Great!

> The pancreas looks a bit "wild" though, @silvandeleemput @fedorov ? ![Bildschirmfoto 2023-12-07 um 14 42 58](https://private-user-images.githubusercontent.com/14541510/288777506-4ecab1e1-334b-4a5f-8a14-588410cbab03.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE5NjM3MTksIm5iZiI6MTcwMTk2MzQxOSwicGF0aCI6Ii8xNDU0MTUxMC8yODg3Nzc1MDYtNGVjYWIxZTEtMzM0Yi00YTVmLThhMTQtNTg4NDEwY2JhYjAzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjA3VDE1MzY1OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU3N2NiOTFjYTkxMjM2Nzg2MWU4MmVkZmVkYmFjODJhYjkwOTFlYTA0N2I1ODQyZDllOWM3OTRkZGY2ZDhmNmYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.KbOzg7rogpof8oS9lAyqOi0OxPChyMMtklWd_sopvuk)

I am not the best to judge, I can maybe ask around next week. I guess you are using 3D Slicer for viewing?

> * typing_extensions conflict needs to be addressed

Yes, as mentioned before, I think this needs to be addressed in the base image. Let's discuss on slack.

> * polluted console out (similar to PICAI); this should be addressed for all algorithms (before we enter the testing phase).

I'll wrap it in a CLI.

> * for final testing, we need your output directory (zipped) and at least one case for each of the segmented structures (the current one only shows the pancreas, pancreatic duct, and bile duct).

So you want one or two cases that have the following two classes as well?
* Cysts
* Renal vein

Do we want to include Arteries and Veins?

silvandeleemput committed:
add cli for running the pdac_detection model

LennyN95 commented:
> I guess you are using 3D Slicer for viewing?

Yes, that's the 3D render of the DicomSEG in Slicer.

> Do we want to include Arteries and Veins?

**EDIT**: I just added them to SegDB, you can use `ARTERY` and `VEIN` as unspecific general IDs.

silvandeleemput committed:
added VEIN,ARTERY rois to output segmentation, cleaned config and runner code

LennyN95 commented on models/gc_nnunet_pancreas/config/default.yml:
```diff
@@ -0,0 +1,35 @@
+    meta: 
+      mod: '%Modality'
+
+  MhaConverter:
```
Note, that `targets` has been renamed to `in_datas` some while ago.
Documentation: https://github.com/MHubAI/models/pull/45#pullrequestreview-1808688218

silvandeleemput replied:
I'll update it.

LennyN95 commented on models/gc_nnunet_pancreas/config/default.yml:
```diff
@@ -0,0 +1,35 @@
+
+  DsegConverter:
+    model_name: 'GC NNUnet Pancreas'
+    source_segs: ['mha:mod=seg']
```
The usage of arrays is deprecated: `source_segs` should be a query string (e.g. `mha:mod=seg`) instead of an array.

silvandeleemput replied:
Ok

LennyN95 commented on models/gc_nnunet_pancreas/config/default.yml:
```diff
@@ -0,0 +1,35 @@
+    targets: [dicom:mod=ct]
+
+  DsegConverter:
+    model_name: 'GC NNUnet Pancreas'
```
I suggest to consistently omit quotes around strings whenever not required (which is only if the string starts with a particular character of meaning in yaml, e.g. `%`, `-`, `[`, ...) to enhance readability.,

silvandeleemput replied:
That's fine, I'll do that.

LennyN95 commented on models/gc_nnunet_pancreas/config/default.yml:
```diff
@@ -0,0 +1,35 @@
+
+  DataOrganizer:
+    targets:
+    - mha:mod=heatmap-->[i:sid]/nnunet_pancreas_heatmap.mha
```
please use the full model name as prefix for all generated output files,
e.g. `gc_nnunet_pancreas_heatmap.mha`.

LennyN95 commented on models/gc_nnunet_pancreas/dockerfiles/Dockerfile:
```diff
@@ -0,0 +1,53 @@
+    cd /opt/algorithm && \
+    git reset --hard e4f4008c6e18e60a79f693448562a340a9252aa8 && \
+    rm -rf /opt/algorithm/.git && \
+    sed -i 's/Path("\/input\/images\/")/Path("\/app")/g' /opt/algorithm/process.py && \
```
Manipulating code inside the Dockerfile goes against our transparency efforts.

As we already use a cli (`utils/cli.py`) to wrap the calls to the python class, can we not just overwrite the class variables from there?

After importing `PDACDetectionContainer`,  from my understanding you, for example, could just set `PDACDetectionContainer.ct_ip_dir = Path("\/app")`.

silvandeleemput replied:
I agree, we discussed this with Natália. We will create a PR with the code changes to the original repository. If that is accepted we will remove the manual code patches.

silvandeleemput replied:
Created a PR for this: https://github.com/DIAGNijmegen/CE-CT_PDAC_AutomaticDetection_nnUnet/pull/5

silvandeleemput replied:
Updated to the latest algorithm/model commit with the merged PR

LennyN95 commented on models/gc_nnunet_pancreas/dockerfiles/Dockerfile:
```diff
@@ -0,0 +1,53 @@
+    git reset --hard e4f4008c6e18e60a79f693448562a340a9252aa8 && \
+    rm -rf /opt/algorithm/.git && \
+    sed -i 's/Path("\/input\/images\/")/Path("\/app")/g' /opt/algorithm/process.py && \
+    sed -i 's/pred_2_np  = sitk\.GetArrayFromImage(pred_2_nii)/pred_2_np  = sitk\.GetArrayFromImage(pred_2_nii)\.astype(np\.uint8)/g' /opt/algorithm/process.py && \
```
These changes likely violate our `as-is` constrains as they appear to modify the code on a functional level. 
is the original code not working (properly) without these modifications? 

This looks to me like updating the original repository would be more suitable (we can discuss in-depth if that should turn out to be problematic).

silvandeleemput replied:
As above. We will create a PR with the code changes to the original repository. If that is accepted we will remove the manual code patches.

silvandeleemput replied:
Created a PR for this: https://github.com/DIAGNijmegen/CE-CT_PDAC_AutomaticDetection_nnUnet/pull/3

silvandeleemput replied:
Updated to the latest algorithm/model commit with the merged PR

LennyN95 commented on models/gc_nnunet_pancreas/meta.json:
```diff
@@ -0,0 +1,129 @@
+    "analyses": {
+      "title": "Analysis",
+      "text": "The study evaluated a medical model's performance for tumor detection by analyzing receiver operating characteristic (ROC) and free-response receiver operating characteristic (FROC) curves, assessing both tumor presence and lesion localization, and compared three configurations using statistical tests and ensemble modeling.",
+      "references": [],
```
Please add references and (if possible but highly encouraged) the most important KPIs from the analysis.

silvandeleemput replied:
We'll do that.

silvandeleemput replied:
Updated the analysis section with a reference to the original paper and a table with the relevant results.

LennyN95 commented on models/gc_nnunet_pancreas/utils/cli.py:
```diff
@@ -0,0 +1,57 @@
+"""
```
Why again did we have the CLI here? 

It'd be beneficial if you could add 1-2 lines of explanation to the file's header why we chose to wrap our code in a cli, what problems that solved etc. to make it easier for future submissions to learn from existing ones.

silvandeleemput replied:
The model uses an nn-UNet internally, so it would produce uncaptured stdout output otherwise. I'll add a few lines of explanation to the file's header.

silvandeleemput replied:
Done

silvandeleemput commented:
Hi, we spoke to Natália today and she says that we should exclude the Bile Duct, Cysts, and Renal vein outputs, as these are not properly validated segmentations. The evaluation of Natália's algorithm was focused on the cancer probability maps and specifically the maximum value of those for the overall case cancer probability. We will add the overall case cancer probability statistic as an output, remove the bile duct cysts and renal veins from the output DICOM segmentation, and add some comments about the limitations of the original output segmentation.

silvandeleemput committed:
add comments to CLI, add clean method and case-level likelihood extraction to runner, add case-level likelihood to config and meta #39

silvandeleemput committed:
meta.json - update analysis section and evaluation data section #39

silvandeleemput committed:
updated model/algorithm version to latest commit, removed manual code patches #39

silvandeleemput committed:
meta.json - match model name #39

silvandeleemput committed:
meta.json - added disclaimer for output segmentation map

silvandeleemput commented:
/review

This model has been updated significantly and needs to be reviewed again. A new test file will be provided that contains all the output segmentations for the segmentation output map and also the new case-level likelihood value. I'll send it via Slack.

LennyN95 commented on models/gc_nnunet_pancreas/meta.json:
```diff
@@ -40,6 +30,17 @@
+      },
+      {
+        "type": "Segmentation",
+        "label": "Segmentation of pancreas related tissues. These segmentation classes were not thoroughly validated, use them on your own risk!",
```
`These segmentation classes were not thoroughly validated, use them on your own risk!`
this is way beyond what should be put in a `label` and should be better put into the limitations section

LennyN95 replied:
Labels should be as short and precise as possible as they're used throughout the UI. You can use `description` for additional details.

silvandeleemput replied:
Thanks for the clarification, I have moved it to the description instead and modified the labels.

silvandeleemput committed:
updated to lastest version of algorithm, changed to output the raw heatmap

silvandeleemput committed:
meta.json - moved segmentation disclaimer to description and modified labels #39

silvandeleemput commented:
[output.zip](https://github.com/MHubAI/models/files/14125409/output.zip)

silvandeleemput commented:
/test

```yaml
sample:
  idc_version: 17
  data:
    - SeriesInstanceUID: 1.3.6.1.4.1.14519.5.2.1.98806944896875131873924657854330253112
      aws: 5475d74e-33db-497e-ace0-00faba1a6a9a
      path: dicom

reference:
  url: https://github.com/MHubAI/models/files/14125409/output.zip
```

---

<details><summary>Test Results (24.02.08_14.58.03_52I9eOKFVS)</summary>

```yaml
checked_files:
- path: /app/test/src/1.3.6.1.4.1.14519.5.2.1.98806944896875131873924657854330253112/nnunet_pancreas.seg.mha
  checker: ImageFileCheck
  findings:
  - label: Dice Score
    description: Dice score between reference and test image
    info: 0.999966
- path: /app/test/src/1.3.6.1.4.1.14519.5.2.1.98806944896875131873924657854330253112/nnunet_pancreas.seg.mha
  checker: SizeCheck
  findings:
  - label: Size Difference
    description: file size is smaller than reference
    info:
      src_size: 83269
      ref_size: 83270
      diff_size: -1
- path: /app/test/src/1.3.6.1.4.1.14519.5.2.1.98806944896875131873924657854330253112/nnunet_pancreas.seg.dcm
  checker: DicomsegContentCheck
- path: /app/test/src/1.3.6.1.4.1.14519.5.2.1.98806944896875131873924657854330253112/nnunet_pancreas_case_level_likelihood.json
  checker: DataFileCheck
- path: /app/test/src/1.3.6.1.4.1.14519.5.2.1.98806944896875131873924657854330253112/nnunet_pancreas_case_level_likelihood.json
  checker: SizeCheck
- path: /app/test/src/1.3.6.1.4.1.14519.5.2.1.98806944896875131873924657854330253112/nnunet_pancreas_heatmap.mha
  checker: ImageFileCheck
  findings:
  - label: Exception
    description: An exception occurred during check
    info: 'could not convert string to float: '''''
- path: /app/test/src/1.3.6.1.4.1.14519.5.2.1.98806944896875131873924657854330253112/nnunet_pancreas_heatmap.mha
  checker: SizeCheck
  findings:
  - label: Size Difference
    description: file size is smaller than reference
    info:
      src_size: 9389151
      ref_size: 9389181
      diff_size: -30
summary:
  files_missing: 0
  files_extra: 0
  checks:
    ImageFileCheck:
      files: 2
      findings:
        Dice Score: 1
        Exception: 1
    SizeCheck:
      files: 3
      findings:
        Size Difference: 2
    DicomsegContentCheck:
      files: 1
    DataFileCheck:
      files: 1
conclusion: false
```

</details>

silvandeleemput committed:
fix dependencies conflict new base image #39

silvandeleemput commented:
[output.zip](https://github.com/MHubAI/models/files/14211021/output.zip)

silvandeleemput commented:
/test

```yaml
sample:
  idc_version: 17.0
  data:
    - SeriesInstanceUID: 1.3.6.1.4.1.14519.5.2.1.237127321134041099456397194653519856815
      aws_url: s3://idc-open-data/be1ba92c-7476-4853-a305-47c8e3e32660/*
      path: dicom

reference:
  url: https://github.com/MHubAI/models/files/14211021/output.zip
```

---

<details><summary>Test Results (24.02.08_19.39.01_bQJenUOogX)</summary>

```yaml
checked_files:
- file: nnunet_pancreas.seg.mha
  path: /app/test/src/1.3.6.1.4.1.14519.5.2.1.237127321134041099456397194653519856815/nnunet_pancreas.seg.mha
  checks:
  - checker: ImageFileCheck
    notes:
    - label: Data Type
      description: Data type of the reference image
      info: uint8
    - label: Dice Score
      description: Dice score between reference and test image
      info: 0.999981
- file: nnunet_pancreas.seg.dcm
  path: /app/test/src/1.3.6.1.4.1.14519.5.2.1.237127321134041099456397194653519856815/nnunet_pancreas.seg.dcm
  checks:
  - checker: DicomsegContentCheck
    notes:
    - label: Segment Count
      description: The number of segments identified in the inspected dicomseg file.
      info: 5
- file: nnunet_pancreas_case_level_likelihood.json
  path: /app/test/src/1.3.6.1.4.1.14519.5.2.1.237127321134041099456397194653519856815/nnunet_pancreas_case_level_likelihood.json
  checks:
  - checker: DataFileCheck
  - checker: SizeCheck
- file: nnunet_pancreas_heatmap.mha
  path: /app/test/src/1.3.6.1.4.1.14519.5.2.1.237127321134041099456397194653519856815/nnunet_pancreas_heatmap.mha
  checks:
  - checker: ImageFileCheck
    notes:
    - label: Data Type
      description: Data type of the reference image
      info: float32
    - label: Image Diff Stat
      description: Statistics of the diff image between src and ref.
      info:
        MIN: -0.000488
        AVE: 0.0
        MAX: 0.000488
        MAE: 0.0
        MSE: 0.0
        DIF: 1082396.0
        NUM: 77856768.0
summary:
  files_missing: 0
  files_extra: 0
  checks:
    ImageFileCheck:
      files: 2
    DicomsegContentCheck:
      files: 1
    DataFileCheck:
      files: 1
    SizeCheck:
      files: 1
conclusion: true
```

</details>

fedorov commented:
@LennyN95 I missed your comment below on slack, and it struck me just now that you changed the format of how AWS path is communicated.

<img width="590" alt="image" src="https://github.com/MHubAI/models/assets/313942/59bb7c16-a7d7-46b7-a94f-13f63db413d3">

If you see my response, I say you need the URL. The UUID is **not** sufficient to locate the folder in AWS.

<img width="525" alt="image" src="https://github.com/MHubAI/models/assets/313942/f28a1610-24f5-4625-806c-b7b928fc6adf">

fedorov commented:
If you want to use just the UUID, you will need: 1) change the label from "aws" to "crdc_series_uuid" - since that is what it is; 2) add a step in your workflow to resolve `crdc_series_uuid` to AWS URL via [`idc-index`](https://github.com/ImagingDataCommons/idc-index). IDC images are spread across several AWS buckets, that is why you cannot use the fixed S3 bucket prefix.

silvandeleemput committed:
meta.json - added version 0.1.0 to details
